test/Microsoft.Azure.Devices.Edge.Test/helpers/DeviceBase.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Helpers
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Test.Common;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Certs;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using NUnit.Framework;
+
+    public class DeviceBase : TestBase
+    {
+        IEdgeDaemon daemon;
+
+        protected CertificateAuthority ca;
+        protected IotHub iotHub;
+
+        [OneTimeSetUp]
+        public async Task BeforeAllAsync()
+        {
+            await Profiler.Run(
+                async () =>
+                {
+                    string agentImage = Context.Current.EdgeAgentImage.Expect(() => new ArgumentException());
+                    string hubImage = Context.Current.EdgeHubImage.Expect(() => new ArgumentException());
+                    (string, string, string) rootCa =
+                        Context.Current.RootCaKeys.Expect(() => new ArgumentException());
+                    Option<Uri> proxy = Context.Current.Proxy;
+                    string deviceId = Context.Current.DeviceId;
+
+                    using (var cts = new CancellationTokenSource(Context.Current.SetupTimeout))
+                    {
+                        CancellationToken token = cts.Token;
+
+                        this.iotHub = new IotHub(
+                            Context.Current.ConnectionString,
+                            Context.Current.EventHubEndpoint,
+                            proxy);
+
+                        this.ca = await CertificateAuthority.CreateAsync(
+                            deviceId,
+                            rootCa,
+                            Context.Current.CaCertScriptPath,
+                            token);
+
+                        this.daemon = OsPlatform.Current.CreateEdgeDaemon(Context.Current.InstallerPath);
+                        await this.daemon.ConfigureAsync(
+                            config =>
+                            {
+                                config.SetCertificates(this.ca.Certificates);
+                                config.Update();
+                                return Task.FromResult(("with edge certificates", Array.Empty<object>()));
+                            },
+                            token);
+
+                        var runtime = new EdgeRuntime(
```

Comment:
is it better to define a property for edge runtime?
```suggestion
                        this.runtime = new EdgeRuntime(
```

Reply:
The device tests don't really touch the Edge device once it is set up. It gets set up with a simple deployment (just the agent and hub modules) before the tests run, and then all the tests do leaf device operations. Since no one needs to use the EdgeRuntime object instance for anything after the inital deployment, I didn't make it a property of the test class.